app/config/services.php

```diff
@@ -2,193 +2,209 @@
- */
+* Loading routes from the routes.php file
+*/
 $di->set('router', function () {
-    return require APP_PATH . '/config/routes.php';
+  return require APP_PATH . '/config/router.php';
 });
 
 /**
- * Flash service with custom CSS classes
- */
+* Flash service with custom CSS classes
+*/
 $di->set('flash', function () {
-    return new Flash([
-        'error' => 'alert alert-danger',
-        'success' => 'alert alert-success',
-        'notice' => 'alert alert-info',
-        'warning' => 'alert alert-warning'
-    ]);
+  return new Flash([
+    'error' => 'alert alert-danger',
+    'success' => 'alert alert-success',
+    'notice' => 'alert alert-info',
+    'warning' => 'alert alert-warning'
+  ]);
 });
 
+// Set up the flash session service
+$di->set(
+  'flashSession',
+  function () {
+    return new FlashSession([
+      'error' => 'alert alert-danger',
+      'success' => 'alert alert-success',
+      'notice' => 'alert alert-info',
+      'warning' => 'alert alert-warning'
+    ]);
+  }
+);
+
 /**
- * Custom authentication component
- */
+* Custom authentication component
+*/
 $di->set('auth', function () {
-    return new Auth();
+  return new Auth();
 });
 
-/**
- * Mail service uses AmazonSES
- */
-$di->set('mail', function () {
-    return new Mail();
-});
 
 /**
- * Setup the private resources, if any, for performance optimization of the ACL.  
- */
+* Setup the private resources, if any, for performance optimization of the ACL.
+*/
 $di->setShared('AclResources', function() {
-    $pr = [];
-    if (is_readable(APP_PATH . '/config/privateResources.php')) {
-        $pr = include APP_PATH . '/config/privateResources.php';
-    }
-    return $pr;
+  $pr = [];
+  if (is_readable(APP_PATH . '/config/privateResources.php')) {
+    $pr = include APP_PATH . '/config/privateResources.php';
+  }
+  return $pr;
 });
 
 /**
- * Access Control List
- * Reads privateResource as an array from the config object.
- */
+* Access Control List
+* Reads privateResource as an array from the config object.
+*/
 $di->set('acl', function () {
-    $acl = new Acl();
-    $pr = $this->getShared('AclResources')->privateResources->toArray();
-    $acl->addPrivateResources($pr);
-    return $acl;
+  $acl = new Acl();
+  $pr = $this->getShared('AclResources')->privateResources->toArray();
+  $acl->addPrivateResources($pr);
+  return $acl;
 });
 
-/**
- * Logger service
- */
-$di->set('logger', function ($filename = null, $format = null) {
-    $config = $this->getConfig();
-
-    $format   = $format ?: $config->get('logger')->format;
-    $filename = trim($filename ?: $config->get('logger')->filename, '\\/');
-    $path     = rtrim($config->get('logger')->path, '\\/') . DIRECTORY_SEPARATOR;
-
-    $formatter = new FormatterLine($format, $config->get('logger')->date);
-    $logger    = new FileLogger($path . $filename);
+/*
+$di->setShared('gravatar', function () {
+// Get Gravatar instance
+$gravatar = new Gravatar([]);
```

Comment:
what is it

Reply:
When a user logs in, the profile shows an avatar from https://gravatar.com/